pyrokinetics/gk_code/GKOutputReaderGENE.py

```diff
@@ -417,40 +470,66 @@ def _set_fluxes(
                         for skip_s in range(data.nspecies + 1):
                             next(nrg_data)
 
-        data["fluxes"] = (coords, fluxes)
+        coords = ["species", "field", "time"]
+        for imoment, moment in enumerate(data["moment"].data):
+            flux = fluxes[:, imoment, :, :] * gene_flux_units[moment]
+            flux = flux.to(local_norm.pyrokinetics)
+            data[moment] = (coords, flux)
+
+        data = data.pint.quantify(pyro_flux_units)
+
         return data
 
     @staticmethod
     def _set_eigenvalues(
         data: xr.Dataset, raw_data: Optional[Any] = None, gk_input: Optional[Any] = None
     ) -> xr.Dataset:
-        if "fields" in data:
+        if "phi" in data:
             return GKOutputReader._set_eigenvalues(data, raw_data, gk_input)
 
-        logging.warn(
-            "'fields' not set in data, falling back to reading 'omega' -- 'eigenvalues' will not be set!"
+        local_norm = data.local_norm
+        pyro_eigval_units = eigenvalues_units(local_norm.pyrokinetics)
+        if gk_input.data["geometry"].get("minor_r", 0.0) == 1.0:
+            gene_eigval_units = eigenvalues_units(local_norm.pyrokinetics)
+        elif gk_input.data["geometry"].get("major_R", 1.0) == 1.0:
+            gene_eigval_units = eigenvalues_units(local_norm.gene)
+        else:
+            raise NotImplementedError(
+                "Pyro does handle cases where neither major_R and minor_r are 1.0"
+            )
+
+        logging.warning(
+            "'phi' not set in data, falling back to reading 'omega' -- 'eigenvalues' will not be set!"
         )
 
         kys = []
-        frequencies = []
-        growth_rates = []
+        mode_frequency = []
+        growth_rate = []
 
         with open(raw_data["omega"], "r") as csv_file:
             omega_data = csv.reader(csv_file, delimiter=" ", skipinitialspace=True)
             for line in omega_data:
-                ky, growth_rate, frequency = line
+                ky, growth, frequency = line
                 kys.append(float(ky))
-                frequencies.append(float(frequency))
-                growth_rates.append(float(growth_rate))
+                mode_frequency.append(float(frequency))
+                growth_rate.append(float(growth))
 
         last_timestep = [data.time.isel(time=-1)]
         coords = {"time": last_timestep, "ky": kys, "kx": [0.0]}
+        dims = coords.keys()
+
+        mode_frequency = mode_frequency * gene_eigval_units["mode_frequency"]
+        mode_frequency = mode_frequency.to(local_norm.pyrokinetics).magnitude
+        growth_rate = growth_rate * gene_eigval_units["growth_rate"]
+        growth_rate = growth_rate.to(local_norm.pyrokinetics).magnitude
```

Comment:
Given that we end up doing this pattern a _lot_, it's probably worth writing something like `convert_dataset()`, like `normalisation.convert_dict`, that iterates over the keys (or a given set of them) in the dataset and does `.to(norm).magnitude` on all of them. Actually, if you also passed in the new units, you'd just have this common pattern of:

```python
# set all the fields in data using units for the local convention
...
return convert_dataset(data, norm, units)
```

And in fact, I guess `units` will contain exactly which keys to convert as well